wsgi.py

```diff
@@ -6,255 +6,348 @@
 import re
 import os
 from flask import Flask, redirect, url_for, request, Response
-#from flask_restful import reqparse
+from flask_httpauth import HTTPBasicAuth
+
+# from flask_restful import reqparse
 
 import sys
 
-from openshift_rolebindings import *
-from openshift_project import *
-from openshift_identity import *
-from openshift_user import *
+from openshift import *
 
 application = Flask(__name__)
+auth = HTTPBasicAuth()
 
-if __name__ != '__main__':
-    gunicorn_logger = logging.getLogger('gunicorn.error')
+if __name__ != "__main__":
+    gunicorn_logger = logging.getLogger("gunicorn.error")
     application.logger.handlers = gunicorn_logger.handlers
     application.logger.setLevel(gunicorn_logger.level)
 
 
-def get_user_token():
-    with open('/var/run/secrets/kubernetes.io/serviceaccount/token', 'r') as file:
+def get_openshift():
+    version = os.environ["OPENSHIFT_VERSION"]
+    url = os.environ["OPENSHIFT_URL"]
+    if version == 3:
+        shift = openshift_3_x(url, version, application.logger)
+        application.logger.info("using Openshift ver 3")
+    else:
+        shift = openshift_4_x(url, version, application.logger)
+        application.logger.info("using Openshift ver 4")
+    with open("/var/run/secrets/kubernetes.io/serviceaccount/token", "r") as file:
         token = file.read()
-        return token
-    return ""
-
-def get_token_and_url():
-    token = get_user_token()
-    openshift_url = os.environ["openshift_url"]
-    return (token, openshift_url)
-
-@application.route("/users/<user_name>/projects/<project_name>/roles/<role>", methods=['GET'])
+        shift.set_token(token)
+        application.logger.info("Attached token to shift")
+        return shift
+    return None
+
+
+@auth.verify_password
+def verify_password(username, password):
+    user_str = ""
+    with open("/app/auth/users", "r") as f:
+        user_str = f.read()
+    user = user_str.split(" ", 1)
+    if username == user[0] and password == user[1]:
+        return username
```

Comment:
@naved001 it's expected to return a username (see https://flask-httpauth.readthedocs.io/en/latest/). This is used so that you have information about *who* authenticated. A simple boolean won't give you that.

Reply:
Thanks! I was a bit confused by this.